sale_procurement_group_by_line/model/sale.py

```diff
@@ -67,6 +67,7 @@ def _action_launch_stock_rule(self, previous_product_uom_qty=False):
             if not group_id:
                 vals = line._prepare_procurement_group_vals()
                 group_id = self.env["procurement.group"].create(vals)
+                line.order_id.procurement_group_id = group_id
```

Comment:
@Jay2163 IMHO, that's not correct as lines in the same order can have a different procurement group.

The way domain is computed on button action should be checked instead